qtfred/src/ui/dialogs/CommandBriefingDialog.cpp

```diff
@@ -0,0 +1,311 @@
+#include "CommandBriefingDialog.h"
+#include "ui_CommandBriefingDialog.h"
+
+#include <globalincs/linklist.h>
+#include <ui/util/SignalBlockers.h>
+#include <QCloseEvent>
+#include <qfiledialog.h>
+
+namespace fso {
+namespace fred {
+namespace dialogs {
+
+
+	CommandBriefingDialog::CommandBriefingDialog(FredView* parent, EditorViewport* viewport)
+	: QDialog(parent), ui(new Ui::CommandBriefingDialog()), _model(new CommandBriefingDialogModel(this, viewport)),
+	_viewport(viewport)
+	{
+		this->setFocus();
+		ui->setupUi(this);
+
+		connect(_model.get(), &AbstractDialogModel::modelChanged, this, &CommandBriefingDialog::updateUI);
+		connect(this, &QDialog::accepted, _model.get(), &CommandBriefingDialogModel::apply);
+		connect(viewport->editor, &Editor::currentObjectChanged, _model.get(), &CommandBriefingDialogModel::apply);
+		connect(viewport->editor, &Editor::objectMarkingChanged, _model.get(), &CommandBriefingDialogModel::apply);
+		connect(this, &QDialog::rejected, _model.get(), &CommandBriefingDialogModel::reject);
+
+		connect(ui->actionChangeTeams,
+			static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+			this,
+			&CommandBriefingDialog::on_actionChangeTeams_clicked);
+
+		connect(ui->actionBriefingTextEditor,
+			&QPlainTextEdit::textChanged,
+			this,
+			&CommandBriefingDialog::briefingTextChanged);
+
+		connect(ui->animationFileName,
+			static_cast<void (QLineEdit::*)(const QString&)>(&QLineEdit::textChanged),
+			this,
+			&CommandBriefingDialog::animationFilenameChanged);
+
+		connect(ui->speechFileName,
+			static_cast<void (QLineEdit::*)(const QString&)>(&QLineEdit::textChanged),
+			this,
+			&CommandBriefingDialog::speechFilenameChanged);
+
+		connect(ui->actionLowResolutionFilenameEdit,
+			static_cast<void (QLineEdit::*)(const QString&)>(&QLineEdit::textChanged),
+			this,
+			&CommandBriefingDialog::lowResolutionFilenameChanged);
+
+		connect(ui->actionHighResolutionFilenameEdit,
+			static_cast<void (QLineEdit::*)(const QString&)>(&QLineEdit::textChanged),
+			this,
+			&CommandBriefingDialog::highResolutionFilenameChanged);
+
+		resize(QDialog::sizeHint());
+
+		_model->requestInitialUpdate();		
+	}
+
+	void CommandBriefingDialog::on_actionPrevStage_clicked() 
+	{ 
+		_model->gotoPreviousStage(); 
+	}
+
+	void CommandBriefingDialog::on_actionNextStage_clicked() 
+	{ 
+		_model->gotoNextStage(); 
+	}
+
+	void CommandBriefingDialog::on_actionAddStage_clicked() 
+	{ 
+		_model->addStage(); 
+	}
+
+	void CommandBriefingDialog::on_actionInsertStage_clicked() 
+	{ 
+		_model->insertStage();  
+	}
+
+	void CommandBriefingDialog::on_actionDeleteStage_clicked() 
+	{ 
+		_model->deleteStage(); 
+	}
+
+	void CommandBriefingDialog::on_actionChangeTeams_clicked ()
+	{
+		// not yet supported
+	}
+
+	void CommandBriefingDialog::on_actionCopyToOtherTeams_clicked()
+	{
+		// not yet supported.
+	}
+
+	void CommandBriefingDialog::on_actionBrowseAnimation_clicked()
```

Comment:
This bit of code is repeated for four functions with only one small difference each time.  It can be separated into its own function and then each onAction can call it.